sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -85,6 +85,13 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper with Join
         } else {
           None
         }
+      case (resExp, r @ DataSourceV2ScanRelation(_, scan: SupportsRuntimeV2Filtering, _, _, _)) =>
```

Comment:
I'm not very sure, is it possible to make `SupportsRuntimeFiltering` extends `SupportsRuntimeV2Filtering`? And inside `SupportsRuntimeFiltering` we translate v2 filter to v1.

Reply:
I know it looks ugly to have duplicate code for both `SupportsRuntimeFiltering` and `SupportsRuntimeV2Filtering`. I thought we will deprecate and remove `SupportsRuntimeFiltering` in the future?

Reply:
removing API is always hard...